app/models/course_user_datum.rb

```diff
@@ -249,6 +249,88 @@ def ggl_cache_key
     "ggl/dua-#{dua}/u-#{self.id}"
   end
 
+  # This method call is used specifically for the purpose of callback style update to watchlist
+  # Need to archive old instances and also add new instances
+  def update_cud_gdu_watchlist_instances
```

Comment:
having a callback function make sense in the course_user_datum, but then it should call functions in the watchlist.rb model to do the actual update. 

Do let me know if this is not clear.